servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/DefaultSingleAddressHttpClientBuilder.java

```diff
@@ -219,6 +228,33 @@ private DefaultSingleAddressHttpClientBuilder(@Nullable final U address,
                     proxyAddress != null ? proxyAddress : builder.address);
         }
 
+        Publisher<? extends ServiceDiscovererEvent<R>> discover(
+                PublisherSource.Processor<Throwable, Throwable> processor) {
+            assert builder.address != null : "Attempted to buildStreaming with an unknown address";
+            return builder.serviceDiscoverer.discover(
+                    proxyAddress != null ? proxyAddress : builder.address)
+                    .retryWhen(retryWithConstantBackoffAndJitter(MAX_VALUE, t -> {
+                        processor.onNext(t);
+                        return t instanceof UnknownHostException;
```

Comment:
Why not retry all SD errors by default?

Reply:
Just to keep existing behavior as-is. But if we will have `retryServiceDiscoveryErrors` on the client builder +1 for retry all by default. Will update.